src/sql/workbench/contrib/notebook/test/electron-browser/cell.test.ts

```diff
@@ -1358,4 +1358,65 @@ suite('Cell Model', function (): void {
 		assert(editMode);
 		assert.strictEqual(editMode, CellEditModes.WYSIWYG, 'Default edit mode should be WYSIWYG.');
 	});
+
+	test('cell should have lastEditMode set to whatever the user edited out of last', async function () {
+		let notebookModel = new NotebookModelStub({
+			name: '',
+			version: '',
+			mimetype: ''
+		});
+		let contents: nb.ICellContents = {
+			cell_type: CellTypes.Markdown,
+			source: '',
+			metadata: {}
+		};
+		let cellModel = factory.createCell(contents, { notebook: notebookModel, isTrusted: false });
+
+		// Non-Editing Preview mode -> showPreview should be true and showMarkdown should be false.
+		assert(cellModel.showPreview, 'showPreview should default to true when not in editMode');
+		assert(!cellModel.showMarkdown, 'showMarkdown should be false when not in editMode');
+
+		let getCurrentCellEditModePromise = () => {
+			return new Promise((resolve, reject) => {
+				setTimeout((error) => reject(error));
```

Comment:
Just remove this completely, we don't need to reject this at all. The test will timeout by itself.